scripts/fetchODCDataFilesUtils.js

```
@@ -0,0 +1,172 @@
+					publishedDate = dateElmContent.substring(dateElmContent.lastIndexOf(matchStartStr) + matchStartStr.length, dateElmContent.lastIndexOf(matchEndStr));
+					publishedDate = module.exports.formatDate(publishedDate);
+					break;
+				}
+			}
+
+			return {
+				downloadUrl: downloadLink,
+				publishedDate: publishedDate
+			};
+		} catch (error) {
+			console.log("Error in getDataSource(): " + error);
+			return {
+				downloadUrl: undefined,
+				publishedDate: undefined
+			};
+		}
+	},
+
+	/**
+	 * Format the date format to YYYY-MM-DD
+	 * @param {String} date A date in any format. In the context of this script, the input format is: November 17, 2020
+	 * @return {String} A date in a format of YYYY-MM-DD. If the input is invalid, return NaN-NaN-NaN
+	 */
+	formatDate: (date) => {
+		const d = new Date(date);
+		let month = "" + (d.getMonth() + 1),
+			day = "" + d.getDate(),
+			year = d.getFullYear();
+
+		month = month.length < 2 ? "0" + month : month;
+		day = day.length < 2 ? "0" + day : day;
+
+		return [year, month, day].join("-");
+	},
+
+	/**
+	 * Check whether a given version of the data is in the repository
+	 * @param {String} dataFileName The name of the file to look for
+	 * @param {String} dataFileDir The folder where all data files are located
+	 * @return {Boolean} true if the file name is already present in the data folder, false if not
+	 */
+	fileNotExists: (dataFileName, dataFileDir) => {
+		let allFiles = fs.readdirSync(dataFileDir);
+		return !allFiles.includes(dataFileName);
+	},
+
+	/**
+	 * Download a file from the given download URL and write into a target local file.
+	 * @param {String} downloadURL The download URL
+	 * @param {String} targetFileLocation The target file location including the path and file name
+	 */
+	downloadDataFile: async (downloadURL, targetFileLocation) => {
+		try {
+			let res = await axios.get(downloadURL);
+			fs.writeFileSync(targetFileLocation, res.data, "utf8");
+		} catch (error) {
+			// do nothing
```

Why swallow error rather than propagating it?